packages/ses/scripts/hermesTransforms.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { parse } from '@babel/parser';
+import babelGenerate from '@agoric/babel-generator';
+import babelTraverse from '@babel/traverse';
+import * as t from '@babel/types';
+
+// @ts-expect-error TS2339 Property 'default' does not exist
+const traverse = babelTraverse.default || babelTraverse;
+const generate = babelGenerate.default || babelGenerate;
+
+const decoder = new TextDecoder('utf-8');
+// @ts-expect-error TS2554 Expected 0 arguments, but got 1
+const encoder = new TextEncoder('utf-8');
+
+const asyncArrowEliminator = {
+  ArrowFunctionExpression(path) {
+    if (path.node.async) {
+      let body = path.node.body;
+
+      path.traverse({
+        ThisExpression(innerPath) {
+          const { start } = innerPath.node.loc;
+          // throw path.buildCodeFrameError("..."); // https://github.com/babel/babel/issues/8617
+          throw Error(
+            `Hermes makeBundle Babel transform doesn't support 'this' keyword in async arrow functions.
+    at this (${path.state.filename}:${start.line}:${start.column})`,
+          );
+        },
+        // No need for an Identifier traversal on nodes matching 'arguments' to error on
+        // Since only non-arrow functions can access the `arguments` array-like object
+      });
+
+      // In case it's a ()=>expression style arrow function
+      if (!t.isBlockStatement(body)) {
+        body = t.blockStatement([t.returnStatement(body)]);
+      }
+
+      const functionExpression = t.functionExpression(
+        null,
+        path.node.params,
+        body,
+        path.node.generator,
+        path.node.async,
+      );
+
+      path.replaceWith(functionExpression);
+    }
+  },
+};
+
+export const hermesTransforms = {
+  mjs: async (
```

Comment:
This will conflict with #2310 which prefers sync transforms.  I'd encourage you to remove `async`, even if TS complains.

Reply:
passing now without `async` 👍 https://github.com/endojs/endo/pull/2334/commits/431bff3d2d18ea5dabf160b3d03d45e31d119f69